[DOCS] Doc-280 AWS golden path with S3 cloud storage and Pandas

Rachel-Reverie opened the pull request:
Changes proposed in this pull request:
- Adds a new tutorial demonstrating every step from installing Great Expectations through Validating Data while using S3 cloud storage and Pandas.

### Definition of Done
Please delete options that are not relevant.

- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

Thank you for submitting!

Rachel-Reverie committed:
DOC-280
Combines installation, setup, and connecting to data (parts 1 & 2 of 4) from other AWS how-to guides into the first half of this tutorial.

Rachel-Reverie committed:
DOC-280
Outlines parts 3 & 4 of 4 as headers.

Rachel-Reverie committed:
DOC-280
Adds content up to part 4.

Rachel-Reverie committed:
DOC-280
Adds all the remaining content.

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=6618&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=6618&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/6618/d8146a17/1dfd05a31b9de2f4f65ffd182eba0ae725b9214a.svg)](https://app.codesee.io/r/reviews?pr=6618&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

Rachel-Reverie committed:
DOC-280
Minor capitalization corrections.

Shinnnyshinshin commented on docs/deployment_patterns/how_to_use_gx_with_aws/how_to_use_gx_with_aws_using_cloud_storage_and_pandas.md:
```diff
@@ -0,0 +1,325 @@
+
+<!-- 1.4.3 Verify that the new Amazon S3 Expectations Store has been added successfully -->
+
+import VerifyS3ExpectationsStoreExists from '@site/docs/guides/setup/configuring_metadata_stores/components_how_to_configure_an_expectation_store_in_amazon_s3/_confirm_that_the_new_expectations_store_has_been_added_by_running_great_expectations_store_list.mdx'
```
For both the `ExpectationStore` and `ValidationStore` steps, I wasn't able to list the previous store (Non-S3) when running `great_expectations store list`, when the document says : "Notice the output contains two Expectation Stores..." 

Not sure if this is just for me, or if the CLI behavior has changed here.

Shinnnyshinshin commented on docs/deployment_patterns/how_to_use_gx_with_aws/how_to_use_gx_with_aws_using_cloud_storage_and_pandas.md:
```diff
@@ -0,0 +1,325 @@
+import CreateAnS3BucketForDataDocs from '@site/docs/guides/setup/configuring_data_docs/components_how_to_host_and_share_data_docs_on_amazon_s3/_create_an_s3_bucket.mdx'
+
+<!-- 1.6.2 Configure your bucket policy to enable appropriate access -->
+import ConfigureYourBucketPolicyToEnableAppropriateAccess from '@site/docs/guides/setup/configuring_data_docs/components_how_to_host_and_share_data_docs_on_amazon_s3/_configure_your_bucket_policy_to_enable_appropriate_access.mdx'
+
```
This portion was new to me, and I'm not sure if I was able to configure it correctly. Is there a possibility that additional resources could be included in the doc? (other than the existing link to AWS)

Shinnnyshinshin commented on docs/deployment_patterns/how_to_use_gx_with_aws/components/_expectation_suite_batch_request_validator_prepare_or_reuse.md:
```diff
@@ -0,0 +1,3 @@
+When we tested our Datasource in step [2.5: Test your new Datasource](how_to_use_gx_with_aws_using_cloud_storage_and_pandas#25-test-your-new-datasource) we also created all of the components we need to begin creating Expectations: A Batch Request to provide sample data we can test our new Expectations against, an empty Expectation Suite to contain our new Expectations, and a Validator to create t ...
+
```
For the `RuntimeDataConnector` and `RuntimeBatchRequest`, I think it might be worth mentioning that we want to `S3 URI` Rather than ARN or Object URL etc.

Rachel-Reverie committed:
Merge branch 'develop' into d/docs/doc-280/aws_golden_path

Rachel-Reverie committed:
Merge branch 'develop' into d/docs/doc-280/aws_golden_path

Rachel-Reverie committed:
DOC-400
- Updates prerequisites admonition to standardized import.

Rachel-Reverie committed:
DOC-400
- Corrects information about `store list` behavior
- Adds clarification about using S3 URI instead of URL when specifying path to a single csv for a batch request.

Shinnnyshinshin commented on docs/components/cli/_store_list_info_active_stores.md:
```diff
@@ -0,0 +1 @@
+Please note that the `great_expectations store list` command will specifically list your *active* Stores, which are the ones specified by `expectations_store_name`, `validations_store_name`, `evaluation_parameter_store_name`, and `checkpoint_store_name` in the file `great_expectations.yml`.  These are the Stores that your Data Context will use by default.
```
❤️

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/cloud/s3/components_pandas/_test_your_new_datasource.mdx:
```diff
@@ -15,6 +15,10 @@ Verify your new <TechnicalTag tag="datasource" text="Datasource" /> by loading d
 
 Add the S3 path to your CSV in the `path` key under `runtime_parameters` in your `BatchRequest`.
 
+:::tip
+The path you will want to use is your S3 URI, not the URL.
+:::
```
Thank you for this @Rachel-Reverie

Shinnnyshinshin reviewed (approved):
Thank you very much for this @Rachel-Reverie. I think the changes look great, and the PR is good to go once the `link-checker` stage passes for you 🙇🏼

Rachel-Reverie committed:
DOC-414
- Adds the guide to the ToC and groups AWS guides under integrations

Rachel-Reverie committed:
DOC-414
- corrects relative paths in links not having file extensions

Rachel-Reverie committed:
Merge branch 'develop' into d/docs/doc-280/aws_golden_path

donaldheppner commented on sidebars.js:
```diff
@@ -276,11 +276,18 @@ module.exports = {
+          label: 'Using Great Expectations with AWS',
+          items: [
+            'deployment_patterns/how_to_use_great_expectations_in_aws_glue',
+            { type: 'doc', id: 'deployment_patterns/how_to_use_gx_with_aws/how_to_use_gx_with_aws_using_cloud_storage_and_pandas' }
```
this renders as "Amazon Web Services" in the TOC, which is needlessly long. I think the nesting is fine, I could see us expanding this section substantially over time (as a good thing!)

Rachel-Reverie committed:
DOC-280
- edits the title of the doc in the ToC to abbreviate AWS

Rachel-Reverie committed:
Merge remote-tracking branch 'origin/d/docs/doc-280/aws_golden_path' into d/docs/doc-280/aws_golden_path

Rachel-Reverie committed:
DOC-280
- reduces "S3 cloud storage" to just "S3"

Rachel-Reverie committed:
Merge branch 'develop' into d/docs/doc-280/aws_golden_path
